### PR TITLE
PIM-7074: Fix Completeness and label filter for OR clauses

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Description (for Contributor and Core Developer)**

Fixes an issue when filtering on both the `label_or_identifier` and `completeness` at the same time.

This issue lays out a more general issue regarding the `SearchQueryBuilder::addShould` function explained in the phpdoc here: https://github.com/akeneo/pim-community-dev/blob/88e69e6ac5763816cf4fdb96c42e988dc8c43d9c/src/Pim/Bundle/CatalogBundle/Elasticsearch/SearchQueryBuilder.php#L81

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Ok
| Added Behats                      | -
| Added integration tests           | Ok
| Changelog updated                 | Ok
| Review and 2 GTM                  | Ok
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: features/Context/CatalogConfigurationContext.php [hunk 1]
```diff
@@ -56,7 +56,7 @@ public function aCatalogConfiguration($catalog)
     {
         $this->initializeReferenceRepository();
 
-        $this->loadCatalog($this->getConfigurationFiles($catalog));
+        $this->loadCatalog($this-getConfigurationFiles($catalog));
```
Comment (Doodoune) [inline features/Context/CatalogConfigurationContext.php]: Why ? - mistake I think

Comment (aRn0D) [inline features/Context/CatalogConfigurationContext.php]: missing `>`

File: src/Pim/Bundle/CatalogBundle/Elasticsearch/SearchQueryBuilder.php [hunk 2]
```diff
@@ -78,6 +78,39 @@ public function addFilter(array $clause)
...
+     *      'A' => 1
+     *    ]
+     *  ],
+     *  [
+     *   'terms' => [
+     *      'A' => 1
+     *    ]
+     *  ]);`
+     *
+     * The resulting logical request looks like this: A = 1 || A = 2 || B = 1 || B = 2 where in fact what the user meant
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Elasticsearch/SearchQueryBuilder.php]: is it on purpose?
do we really use the `should` clause somewhere? should be use it?

Comment (SamirBoulil) [reply src/Pim/Bundle/CatalogBundle/Elasticsearch/SearchQueryBuilder.php]: I don't think we should prevent people from adding should clauses in the API, but yes in the search context using the addShould clauses can damage the search results.

**Pull request merged**